### ISSUE TITLE
Can the platformmenubar widget add a state field

### ISSUE THREAD

**Issue by JDongKhan** [flutter/flutter#105487]
Can the platformmenubar widget add a state field? tks!

[✓] Flutter (Channel stable, 3.0.0, on macOS 12.1 21C52 darwin-x64, locale zh-Hans-CN)
    • Flutter version 3.0.0 at /Users/jd/fvm/versions/3.0.0
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (4 weeks ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

![Snip20220607_41](https://user-images.githubusercontent.com/4929805/172277381-64f2e324-b937-4488-8066-eec984ff6582.png)
![image](https://user-images.githubusercontent.com/4929805/172277446-7ac3d95d-92d6-409c-be4b-4db112e6cdb5.png)

**Comment by HansMuller** [comment]
CC @gspencergoog

**Comment by gspencergoog** [comment]
Yes, we are planning on adding checkbox menu items, we just haven't gotten to it yet.

**Comment by driftwoodstudio** [comment]
"Yes, we are planning on adding checkbox menu items, we just haven't gotten to it yet."

Closing in on three years now since that comment. 

This feature is pretty important for "View" menu items that show/hide parts of the UI (toggle actions)

**Comment by PrimaryFeather** [comment]
Since the macOS target has been stable since 2022, and given that even iPadOS now includes a system menu bar, I think it’s becoming increasingly important to support both on/off states and menu item icons.

These are common UX patterns on both macOS and iPadOS, and their absence makes Flutter desktop apps feel less native. Implementing them would help Flutter apps better align with platform conventions – and, more importantly, user expectations.

**Comment by mstfkhazaal** [comment]
Any update for this?
